Sources/Sentry/SentryOptions.m

```diff
@@ -25,14 +33,16 @@ - (instancetype)init
         self.maxBreadcrumbs = defaultMaxBreadcrumbs;
         self.maxCacheItems = 30;
         self.integrations = SentryOptions.defaultIntegrations;
-        self.sampleRate = @1;
+        _defaultSampleRate = @1;
+        self.sampleRate = _defaultSampleRate;
         self.enableAutoSessionTracking = YES;
         self.enableOutOfMemoryTracking = YES;
         self.sessionTrackingIntervalMillis = [@30000 unsignedIntValue];
         self.attachStacktrace = YES;
         self.maxAttachmentSize = 20 * 1024 * 1024;
         self.sendDefaultPii = NO;
-        self.tracesSampleRate = @0;
+        _defaultTracesSampleRate = @0;
```

Comment:
I believe the default here should be `nil`.

`0` means: Don't randomly create new transactions but if a trace has already been sampled, then honor that sampling decision.